File: Java/백준/Silver/15651. N과 M （3）/N과 M （3）.java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {

        Scanner s = new Scanner(System.in);

        int N = s.nextInt();
        int M = s.nextInt();

        int[] arr = new int[M];
        StringBuilder sb = new StringBuilder();
        backtrack(N, M, 0, arr, sb);
        System.out.println(sb.toString());

    }

    public static void backtrack(int n, int m, int depth, int[] arr, StringBuilder sb) {
        if (depth == m) {
            for (int i = 0; i < m; i++) {
                sb.append(arr[i]).append(" ");
            }
            sb.append("\n");
            return;
        }
        for (int i = 1; i <= n; i++) {
            arr[depth] = i;
            backtrack(n, m, depth + 1, arr, sb);
        }
    }
}

File: C11/백준/Silver/15651. N과 M （3）/N과 M （3）.c
#include <stdio.h>

void backtrack(int n, int m, int depth, int arr[]){
	if(depth == m){
		for(int i = 0; i < m; i++){
			printf("%d ", arr[i]);
		}
		printf("\n");
		return;
	}
	for(int i = 1; i <= n; i++){
		arr[depth] = i;
		backtrack(n, m, depth + 1, arr);
	}
}

int main(void){
	
	int N, M;
	scanf("%d %d", &N, &M);
	
	int arr[10];
	
	backtrack(N, M, 0, arr);
	
	return 0;
}